src/fragments/lib-v1/ssr/nextjs/getting-started.mdx

```diff
@@ -0,0 +1,338 @@
+    nextServerContext: { request, response },
+    operation: async (contextSpec) => {
+      try {
+        const session = await fetchAuthSession(contextSpec, {});
+        return session.tokens !== undefined;
+      } catch (error) {
+        console.log(error);
+        return false;
+      }
+    },
+  });
+
+  if (authenticated) {
+    return response;
+  }
+
+  return NextResponse.redirect(new URL('/sign-in', request.url));
+}
+
+export const config = {
+  matcher: [
+    /*
+     * Match all request paths except for the ones starting with:
+     * - api (API routes)
+     * - _next/static (static files)
+     * - _next/image (image optimization files)
+     * - favicon.ico (favicon file)
+     */
+    '/((?!api|_next/static|_next/image|favicon.ico|sign-in).*)',
+  ],
+};
+```
+
+In this example, if the incoming request is not associated with a valid user
+session the request will be redirected to the `/sign-in` route.
+
+<Callout>
+
+**NOTE:** When calling `fetchAuthSession` with a `response` context, it will send the
+refreshed tokens (if any) back to the client via the `Set-Cookie` header in the response.
+
+</Callout>
+
+## Using Amplify Category APIs as a Data Source on The Server Side
+
+In general, to use Amplify APIs on the server in your Next.js app, you will
+need to:
+
+1. Import the API from the `/server` sub path.
+2. Using the `runWithAmplifyServerContext` helper function exported from
+`@aws-amplify/adapter-nextjs` to call the Amplify API in an isolated server context.
+
+<Callout>
+
+**NOTE:** A subset of Amplify APIs can now be called on the server side of a Next.js
+app. These APIs are exported from the `/server` sub paths. See
+[the full list](/lib-v1/ssr/nextjs/q/platform/js/#amplify-api-nextjs-server-support)
+of supported APIs.
+
+</Callout>
+
+### With Next.js App Router
+
+#### In React Server Component
+
+##### Dynamic Rendering
+
+Dynamic rendering is based on a user session extracted from an incoming request.
+
+```typescript
+import { cookies } from 'next/headers';
+import { runWithAmplifyServerContext } from '@aws-amplify/adapter-nextjs';
+import { getCurrentUser } from '@aws-amplify/auth/server';
+
+// This page always dynamically renders per request
+export const dynamic = 'force-dynamic';
+
+export default async function AuthGetCurrentUserServer() {
+  try {
+    const currentUser = await runWithAmplifyServerContext({
+      nextServerContext: { cookies },
+      operation: (contextSpec) => getCurrentUser(contextSpec),
+    });
+
+    return (
+      <AuthFetchResult
+        description="The API is called on the server side."
+        data={currentUser}
+      />
+    );
+  } catch (error) {
+    console.error(error);
+    return <p>Something went wrong...</p>;
+  }
+}
+```
+
+##### Static Rendering
+
+Static rendering doesn’t require a user session, so you can specify the
+`nextServerContext` parameter as `null`. This is useful for some use cases,
```

Comment:
does it matter if it's `null` or `undefined`? curious if this was intended to be explicitly set. If so, would love to have the explanation below the code snippet to inform cutomers of potential pitfalls.